pkg/controller/controller.go

```diff
@@ -283,8 +284,10 @@ func NewSelector(matchesFunc func(labels labels.Labels) bool) selector.Selector
 }
 
 func (c *Controller) Stop(ctx context.Context) {
-	c.collector.Stop(ctx)
-	close(c.stopped)
+	c.stopOnce.Do(func() {
+		c.collector.Stop(ctx)
+		close(c.stopped)
+	})
```

Comment:
I'm fine with that but are you sure closing it multiple times isn't a bug in your code?

Reply:
The problem is that we can shut down the controller without exiting the rest of the program. Then if we send a SIGTERM later it will try to close the channel again.